presto-parquet/src/main/java/io/prestosql/parquet/reader/ShortDecimalColumnReader.java

```diff
@@ -47,54 +52,104 @@
     protected void readValue(BlockBuilder blockBuilder, Type prestoType)
     {
         if (definitionLevel == columnDescriptor.getMaxDefinitionLevel()) {
-            long decimalValue;
+            if (!((prestoType instanceof DecimalType) || isNumberType(prestoType))) {
+                throw new ParquetDecodingException(format("Unsupported Presto column type (%s) for Parquet column (%s)", prestoType, columnDescriptor));
+            }
+
+            long value;
+
             // When decimals are encoded with primitive types Parquet stores unscaled values
             if (columnDescriptor.getPrimitiveType().getPrimitiveTypeName() == INT32) {
-                decimalValue = valuesReader.readInteger();
+                value = valuesReader.readInteger();
             }
             else if (columnDescriptor.getPrimitiveType().getPrimitiveTypeName() == INT64) {
-                decimalValue = valuesReader.readLong();
+                value = valuesReader.readLong();
             }
             else {
-                decimalValue = getShortDecimalValue(valuesReader.readBytes().getBytes());
+                value = getShortDecimalValue(valuesReader.readBytes().getBytes());
             }
 
-            if (isShortDecimal(prestoType)) {
-                validateDecimal((DecimalType) prestoType);
-                // TODO: validate that value fits Presto decimal precision
-                prestoType.writeLong(blockBuilder, decimalValue);
-            }
-            else if (isLongDecimal(prestoType)) {
-                validateDecimal((DecimalType) prestoType);
-                // TODO: validate that value fits Presto decimal precision
-                prestoType.writeSlice(blockBuilder, unscaledDecimal(decimalValue));
+            DecimalType prestoDecimalType;
+
+            if (prestoType instanceof DecimalType) {
+                prestoDecimalType = (DecimalType) prestoType;
             }
-            else if (prestoType.equals(TINYINT) || prestoType.equals(SMALLINT) || prestoType.equals(INTEGER) || prestoType.equals(BIGINT)) {
+            else {
+                // for numeric types convert decimal to unscaled long
+                prestoDecimalType = DecimalType.createDecimalType(MAX_SHORT_PRECISION, 0);
```

Comment:
I don't like fake information, because this is error prone.
Fortunately, the result of this assignment is never used.
You can restructure the code slightly and remove the fake.